### DISCUSSION TITLE
dynamically assign an image?

### DISCUSSION THREAD

**Discussion by erwin** [muesli/deckmaster#103] [Q&A]
If I have 5 different volume level icons and want to assign the icon most closely matching the current volume (probably by parsing pactl output) is this feasible with `deckmaster` right now?

I installed it, and have played with it a bit. The defaults are super interesting!

I saw that a `command` output can be assigned to a button, but can a buttons image be reassigned?

Or is there a API for adding new command types, and just writing the command in go instead of shell?

**Comment by falko-apheris** [comment]
I was missing this too so i added a `icon_command`
```
# Docker button
[[keys]]
  index = 7
  [keys.widget]
    id = "command"
    interval = 5000
    [keys.widget.config]
      icon_command = "bun decks/docker/docker-render.js badge"
      command = ""
  [keys.action]
    deck = "docker/docker.deck"
```
https://github.com/select/deckblaster
